pipeline/load.py

```diff
@@ -0,0 +1,220 @@
+
+
+def add_magtype_to_db(conn: connection, cursor: cursor, magtype_value: str) -> int:
+    """Adds the magType value to the RDS"""
+    query = """INSERT INTO magtypes (magtype_value) VALUES (%s) RETURNING magtype_id"""
+    return execute_insert(conn, cursor, query, (magtype_value,), MAGTYPE_ID)
+
+
+def get_magtype_id(conn: connection, cursor: cursor, magtype_value: str, all_magtypes: dict) -> int:
+    """Gets the magtype_id for the provided magType value"""
+    return get_or_add_id(magtype_value, all_magtypes, lambda value_to_add: add_magtype_to_db(conn, cursor, value_to_add))
+
+
+def add_type_to_db(conn: connection, cursor: cursor, earthquake_type: str) -> int:
+    """Adds the earthquake type to the RDS"""
+    query = """INSERT INTO types (type_value) VALUES (%s) RETURNING type_id"""
+    return execute_insert(conn, cursor, query, (earthquake_type,), TYPE_ID)
+
+
+def get_type_id(conn: connection, cursor: cursor, earthquake_type: str, all_types: dict) -> int:
+    """Gets the type_id for the provided earthquake type value"""
+    return get_or_add_id(earthquake_type, all_types, lambda value_to_add: add_type_to_db(conn, cursor, value_to_add))
+
+
+def add_earthquake_data_to_rds(conn: connection, cursor: cursor, earthquake_data: list[dict], all_alerts: dict, all_statuses: dict, all_networks: dict, all_magtypes: dict, all_types: dict) -> None:
+    """Adds the provided data to the 'earthquakes' table"""
+    try:
+        for earthquake in earthquake_data:
+            alert_id = all_alerts.get(earthquake.get(ALERT))
+            status_id = all_statuses.get(earthquake["status"])
+            network_id = get_network_id(
+                conn, cursor, earthquake["network"], all_networks)
+            magtype_id = get_magtype_id(
+                conn, cursor, earthquake["magtype"], all_magtypes)
+            type_id = get_type_id(
+                conn, cursor, earthquake["earthquake_type"], all_types)
+
+            cursor.execute("""INSERT INTO earthquakes (earthquake_id, alert_id, status_id, network_id, magtype_id, type_id, magnitude, lon, lat, depth, time, felt, cdi, mmi, significance, nst, dmin, gap, title)
+                            VALUES (%s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s)
+                            """, (earthquake["earthquake_id"], alert_id, status_id, network_id, magtype_id, type_id,
+                                  earthquake["magnitude"], earthquake["lon"], earthquake["lat"], earthquake["depth"],
+                                  earthquake["time"], earthquake["felt"], earthquake["cdi"], earthquake["mmi"],
+                                  earthquake["significance"], earthquake["nst"], earthquake["dmin"], earthquake["gap"], earthquake["title"]))
+
+            conn.commit()
+            logging.info(f"Successfully added earthquake {
+                         earthquake['earthquake_id']} to the database")
```

Comment:
may be good to add these just in case:

cursor.close()
conn.close()

Reply:
Good shout - added this